doc/cprover-manual/api.md

```diff
@@ -211,6 +211,20 @@ returns true when it is safe to do both.  These predicates can be given an
 optional size; when the size argument is not given, the size of the subtype
 (which must not be **void**) of the pointer type is used.
 
+#### \_\_CPROVER\_havoc\_object, \_\_CPROVER\_havoc\_object\_slice
+
+
+This function requires a valid pointer and updates all the bytes of the underlying object with nondeterministic values.
+```C
+void __CPROVER_havoc_object(void *p);
```

Comment:
When you say "object", what do you mean?
For example, should this assertion pass or fail?
```C
struct foo {
  int x;
  int y;
};

struct foo thefoo = {.x = 1; .y = 1|;
int* p = &thefoo.y;
__CPROVER_havoc_object(p);
assert(thefoo.x == 1);
```

Reply:
the object in that case is the blob of memory that holds the whole struct `thefoo` so this assertion should fail.
on the other it passes when using `__CPROVER_havoc_object_slice(p, sizeof(*p))`

Reply:
This seems like it could lead to a foot gun:

```C
int foo_stub(int* p) {
   __CPROVER_havoc_object(p);
   return *p;
}

struct foo { int x; int y; };

int main() [
  struct foo thefoo = {.x = 1; .y = 12};
  int v = foo_stub(&thefoo.y);
  assert(thefoo.x == 1);
  assert(thefoo.y == v);
}
```

I think the user would expect that foo_stub wouldn't modify memory which it doesn't have a valid pointer to.  Maybe we need a different name: 

```
__CPROVER_havoc_entire_allocation()
```

Reply:
Agreed, we could name `__CPROVER_havoc_object` differently but it was introduced some time ago so this could be a breaking change for existing users of the primitive.

Reply:
Can we just add a more explicit warning to the document about havoc_object that it will havoc enclosing  structs.